File: XMLCheck/Handler/CheckRulesXML01/Check_MA_DANTOC.cs
using XMLCheck.Handler.CheckRulesXML1;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML01
{
    public class Check_MA_DANTOC:Xml1RuleBase
    {
        public override string Key => "MA_DANTOC";
        public override ErrorDetails? Check(Xml1Model model, HoSoContext context)
        {

            if (string.IsNullOrWhiteSpace(model.MA_DANTOC))
            {
                return Error($"{Key} không được rỗng");
            }
            if (!DictionaryCache.MaDanToc.ContainsKey(model.MA_DANTOC))
            {
                return Error($"{Key} không tồn tại trong danh mục");
            }
            return null;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml1/Check_MA_DANTOC.ts
import { Xml1RuleBase } from './xml1RuleBase';
import { Xml1Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { DictionaryCacheInstance } from '../../dictionary.cache';

export class Check_MA_DANTOC extends Xml1RuleBase
    {
  constructor() {
    super({
      ruleId: 'RULE_XML1_MA_DANTOC_01',
      xmlType: 'XML1',
      field: 'MA_DANTOC',
      severity: 'ERROR',
      description: 'Kiểm tra quy định cho trường MA_DANTOC',
      errorMessage: 'MA_DANTOC không được rỗng',
      reference: 'QĐ 3176/QĐ-BYT',
    });
  }

        
        public check(model: Xml1Model, context: HoSoContext): ErrorDetails | null
        {

            if (!model.MA_DANTOC || !model.MA_DANTOC.trim())
            {
                return this.error(`${this.key} không được rỗng`);
            }
            if (!(model.MA_DANTOC in DictionaryCacheInstance.maDanToc))
            {
                return this.error(`${this.key} không tồn tại trong danh mục`);
            }
            return null;
        }
    }
